### DISCUSSION TITLE
Ability to disable model scoping for route

### DISCUSSION THREAD

**Discussion by yakatz** [laravel/framework#39643] [Ideas]
I have an application for first-responders that sends SMS notifications and has a unique link for each responder to click on to acknowledge they have received the message. The initial route definition looks like this:
```php
Route::get('sms/{voicemail:smsKey}-{user:smsKey}', [VoicemailController::class, 'showSms'])
    ->name('voicemail.sms');
```

The problem is that there isn't really a relationship between the `Voicemail` and `User` models, those are just two separate pieces of data that I need to know. There doesn't seem to be a way to tell the router not to scope the second item to the first one. Explicit declaration of intent to scope was added in #39440. What about explicit declaration of intent to *not scope*?
```php
Route::get('sms/{voicemail:smsKey}-{user:smsKey}', [VoicemailController::class, 'showSms'])
    ->name('voicemail.sms')
    ->noScopeBindings();
```

If you want to make it more advanced, you could specify which models should not be scoped:
```php
Route::get('sms/{voicemail:smsKey}-{user:smsKey}', [VoicemailController::class, 'showSms'])
    ->name('voicemail.sms')
    ->noScopeBindings(['user']);
```

---

The (ridiculous) workaround:
```php
Route::bind('userSmsKey', function ($value) {
    return (new User)->resolveRouteBinding($value, 'smsKey');
});
````
```php
Route::get('sms/{voicemail:smsKey}-{userSmsKey}', [VoicemailController::class, 'showSms'])
    ->name('voicemail.sms');
```

**Comment by sergio-agosti** [comment]
I'm struggling with this, too. It's so confusing that just by using a custom column for the models, scoped bindings are silently turned on. I think it would be nice to at least have an option to opt-out of this behaviour (ideally, also, globally). With L9 we now have the `scopeBindings()` route method which would allow to re-enable this behaviour on demand, making it much more obvious and avoiding confusion.

**Comment by stevethomas** [comment]
I've opened up a PR to try and resolve this in Laravel 9: https://github.com/laravel/framework/pull/44528
